src/api.rs

```diff
@@ -1079,6 +1101,10 @@ impl<T: Pixel> ContextInner<T> {
       output_frameno_in_gop,
       self.gop_input_frameno_start[&output_frameno]
     );
+    let frames_needed = input_frameno + self.inter_cfg.keyframe_lookahead_distance();
+    if frames_needed > self.next_lookahead_frame && self.needs_more_frames(frames_needed) {
```

Comment:
What happens when we are in flushing state?

Reply:
This works as if we reached the end of the video, because the `limit` gets set at the start of flushing.